### ISSUE TITLE
Add loading bar to enhance UX for large decks

### ISSUE THREAD

**Issue by adelq** [adelq/anki_hierarchy#4]
Currently, when running this on large decks, it appears to hang. Adding a progress bar to indicate status would be much better.

**Comment by seh5408** [comment]
Hey,

I love this add on, and it was working for me but seems to have stopped? Not sure - I'm using it on really large decks so I'm wondering if it is maybe hanging as you describe. It just says "Waiting for editing to finish." Should I just give it more time? Thanks!

**Comment by huttarl** [comment]
Similar experience: Need visual status indication.

When I ran "Tools > Convert subdecks to tags", the main window immediately displayed the "Waiting for editing to finish" message with the "Resume Now" button. (There may have been another window for a brief moment ... if so it disappeared very quickly.) I waited a while, but nothing seemed to be happening. I checked for other Anki windows, but found none. I checked the CPU activity. There was no obvious load from Anki. I didn't want to click "Resume Now" and mess things up, if the operation was in progress. But there was no indication that it was still churning, and no indication that it was finished. Finally I decided it must be done, and clicked "Resume Now." Apparently it was done. Why then did it continue to display "Waiting for editing to finish"?

Letting the user know that the operation is in progress, or even just letting them know that the operation is finished, would be a big help here.
